### PR TITLE
auto create pull request for cleanup job

### PR DESCRIPTION
**Pull request by sabrina-datadog**
### What does this PR do?
Adds logic in the `release.cleanup` job to auto create the last stable update pull request ([BARX-674](https://datadoghq.atlassian.net/browse/BARX-674))

### Motivation
Reduces the manual work involved in updating the stable version after each release.

### Describe how to test/QA your changes
Ran `inv release.cleanup` job locally with version `7.60.0` (version that isn't stable yet) and checked if a PR is created properly.

Here is [the test PR](https://github.com/DataDog/datadog-agent/pull/31059):

image

### Additional Notes

[BARX-674]: https://datadoghq.atlassian.net/browse/BARX-674?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tasks/libs/ciproviders/github_api.py [hunk 1]
```diff
@@ -491,9 +493,11 @@ def create_release_pr(title, base_branch, target_branch, version, changelog_pr=F
         "qa/no-code-change",
         "team/agent-delivery",
         "team/agent-release-management",
-        "category/release_operations",
     ]
 
+    if not laststable_pr:
+        labels.append("category/release_operations")
```
Comment (kacper-murzyn) [inline tasks/libs/ciproviders/github_api.py:499]: I think that we actually don't need this label any more. It used to be required in the past but we stopped using it recently. So the logic behind the `laststable_pr` will not be needed.

File: tasks/libs/ciproviders/github_api.py [hunk 2]
```diff
@@ -451,13 +451,15 @@ def get_user_query(login):
...
+def create_release_pr(
+    title, base_branch, target_branch, version, changelog_pr=False, laststable_pr=False, milestone=None
+):
     print(color_message("Creating PR", "bold"))
 
     github = GithubAPI(repository=GITHUB_REPO_NAME)
 
     # Find milestone based on what the next final version is. If the milestone does not exist, fail.
-    milestone_name = str(version)
+    milestone_name = milestone if milestone else str(version)
```
Comment (CelianR) [inline tasks/libs/ciproviders/github_api.py:462]: NIT:
```suggestion
    milestone_name = milestone or str(version)
```

File: tasks/release.py [hunk 3]
```diff
@@ -749,7 +751,39 @@ def cleanup(ctx):
...
+        raise Exit(
+            color_message(
+                f"Could not create commit. Please commit manually with:\ngit commit -m {commit_message}\n, push the {cleanup_branch} branch and then open a PR against {main_branch}.",
+                "red",
+            ),
+            code=1,
+        )
+
+    res = ctx.run(f"git push --set-upstream origin {cleanup_branch}", warn=True)
+    if res.exited is None or res.exited > 0:
```
Comment (CelianR) [inline tasks/release.py:774]: NIT:
```suggestion
    if not ctx.run(f"git push --set-upstream origin {cleanup_branch}", warn=True):
```

File: tasks/release.py [hunk 4]
```diff
@@ -749,7 +751,39 @@ def cleanup(ctx):
...
+    current_milestone = _update_last_stable(ctx, version)
+
+    # create pull request to update last stable version
+    main_branch = "main"
+    cleanup_branch = f"release/{version}-cleanup"
+    ctx.run(f"git checkout -b {cleanup_branch}")
+    ctx.run("git add release.json")
+
+    commit_message = f"Update last_stable to {version}"
+    ok = try_git_command(ctx, f'git commit -m "{commit_message}"')
```
Comment (CelianR) [inline tasks/release.py:763]: NIT: (to avoid expansions)
```suggestion
    ok = try_git_command(ctx, f"git commit -m '{commit_message}'")
```

**Review by CelianR — APPROVED** [review]

**Review by KevinFairise2 — APPROVED** [review]

File: tasks/release.py [hunk 5]
```diff
@@ -724,7 +724,7 @@ def create_release_branches(ctx, base_directory="~/dd", major_versions="6,7", up
```
Comment (pducolin) [inline tasks/release.py:724]: **❓ question**
Should this file be owned by @DataDog/agent-delivery only?

Comment (sabrina-datadog) [reply tasks/release.py:724]: Yes, that makes sense

**Pull request merged**